src/sql/workbench/contrib/editorReplacement/common/editorReplacerContribution.ts

```diff
@@ -60,7 +60,7 @@ export class EditorReplacementContribution implements IWorkbenchContribution {
 		if (!language) {
 			// Attempt to use extension or extension of modified input (if in diff editor)
 			// remove the .
-			language = editor instanceof DiffEditorInput ? path.extname(editor.modifiedInput.resource.toString()).slice(1) : path.extname(editor.resource.toString()).slice(1);
+			language = editor instanceof DiffEditorInput ? path.extname(editor.modifiedInput.resource.fsPath).slice(1) : path.extname(editor.resource.toString()).slice(1);
```

Comment:
What about the non-DiffEditorInput case? It's clearly been working fine so far - but I'm wondering if this stuff @VasuBhog is doing with parameters is going to make it so we start having queries and fragments on editor resources...

The fix seems fine as a targeted fix, but it would be great to follow up and see if we want to do the same for the other case too.

Reply:
Reasonable question. https://github.com/microsoft/azuredatastudio/issues/15076 to track. Will work with Alan as well for the sql case.